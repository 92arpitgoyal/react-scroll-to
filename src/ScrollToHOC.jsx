import React from "react";
import getDisplayName from "./utilities/getDisplayName";
import ScrollTo from "./ScrollTo";

/**
 * Higher Order Component version of the ScrollTo.
 * Injects a prop named scroll that is a function that
 * takes an (x, y) coordinate to scroll to. [ie. props.scrollTo(0, 500)]
 */
function ScrollToHOC(Component) {
  const WrappedComponent = props => (
    <ScrollTo>
      {({ scrollTo, scrollById }) => (
        <Component {...props} scrollTo={scrollTo} scrollById={scrollById} />
      )}
    </ScrollTo>
  );
  WrappedComponent.displayName = `WithScrollToHOC(${getDisplayName(
    Component
  )})`;

  return WrappedComponent;
}

export default ScrollToHOC;
