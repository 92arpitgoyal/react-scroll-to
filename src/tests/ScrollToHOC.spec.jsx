import React from "react";
import { render, Simulate } from "react-testing-library";
import ScrollToHOC from "../ScrollToHOC";

beforeEach(() => {
  window.scroll = jest.fn();
});

describe("Test HOC.", () => {
  it("Should render the functional children.", () => {
    const TestComponent = () => <div>test</div>;
    TestComponent.displayName = "test";

    const WrappedComponent = ScrollToHOC(TestComponent);
    const { container } = render(<WrappedComponent />);

    expect(container).toMatchSnapshot();
  });

  it("Should call window.scroll.", () => {
    const TestComponent = props => (
      <button onClick={() => props.scrollTo(100, 200)}>test</button>
    );
    TestComponent.displayName = "test";

    const WrappedComponent = ScrollToHOC(TestComponent);
    const { getByText } = render(<WrappedComponent />);

    Simulate.click(getByText("test"));

    expect(window.scroll).toHaveBeenCalledTimes(1);
    expect(window.scroll.mock.calls[0]).toEqual([100, 200]);
  });

  it("Should call scrollById.", () => {
    const WrappedComponent = ScrollToHOC(props => {
      return (
        <div>
          <div id="foo" />
          <button onClick={() => props.scrollById("foo", 100, 200)}>
            test
          </button>
        </div>
      );
    });

    const { getByText, container } = render(<WrappedComponent />);

    Simulate.click(getByText("test"));

    expect(container).toMatchSnapshot();
  });
});
