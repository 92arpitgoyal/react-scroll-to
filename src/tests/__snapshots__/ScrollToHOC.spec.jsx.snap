// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HOC. Should call scrollById. 1`] = `
<WithScrollToHOC(Unknown)>
  <ScrollTo>
    <Component
      scrollById={[Function]}
      scrollTo={[Function]}
    >
      <div>
        <div
          id="foo"
        />
        <button
          onClick={[Function]}
        >
          test
        </button>
      </div>
    </Component>
  </ScrollTo>
</WithScrollToHOC(Unknown)>
`;

exports[`Test HOC. Should render the functional children. 1`] = `
<WithScrollToHOC(test)>
  <ScrollTo>
    <test
      scrollById={[Function]}
      scrollTo={[Function]}
    >
      <div>
        test
      </div>
    </test>
  </ScrollTo>
</WithScrollToHOC(test)>
`;
